protocol/x/affiliates/keeper/keeper_test.go

```diff
@@ -14,3 +20,193 @@ func TestLogger(t *testing.T) {
 	logger := tApp.App.AffiliatesKeeper.Logger(ctx)
 	require.NotNil(t, logger)
 }
+
+func TestRegisterAffiliate(t *testing.T) {
+	tApp := testapp.NewTestAppBuilder(t).Build()
+	ctx := tApp.InitChain()
+	k := tApp.App.AffiliatesKeeper
+
+	tests := []struct {
+		name        string
+		referee     string
+		affiliate   string
+		expectError bool
+	}{
+		{
+			name:        "Register new affiliate",
+			referee:     "referee1",
+			affiliate:   "affiliate1",
+			expectError: false,
+		},
+		{
+			name:        "Register existing referee",
+			referee:     "referee1",
+			affiliate:   "affiliate2",
+			expectError: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			err := k.RegisterAffiliate(ctx, tc.referee, tc.affiliate)
+			if tc.expectError {
+				require.Error(t, err)
```

Comment:
Return early here to avoid indent: https://go.dev/wiki/CodeReviewComments#indent-error-flow